expression/constant_fold.go

```diff
@@ -78,6 +80,58 @@ func ifNullFoldHandler(expr *ScalarFunction) (Expression, bool) {
 	return expr, isDeferredConst
 }
 
+func caseWhenHandler(expr *ScalarFunction) (Expression, bool) {
+	args, l := expr.GetArgs(), len(expr.GetArgs())
+	isFirstCondition := true
+	isDeferredConst := false
+	for i := 0; i < l-1; i += 2 {
+		foldedArg, isDeferred := foldConstant(args[i])
+		expr.GetArgs()[i] = foldedArg
+		isDeferredConst = isDeferredConst || isDeferred
+		if _, isConst := foldedArg.(*Constant); isConst {
+			// If the condition is const and true, and the previous conditions
+			// has no expr, then the folded execution body is returned, otherwise
+			// the arguments of the casewhen are folded and replaced.
+			condition, isNull, err := args[i].EvalInt(expr.GetCtx(), chunk.Row{})
+			if err != nil {
+				return expr, false
+			}
+			if isFirstCondition && condition != 0 && !isNull {
+				return retProcess(args[i+1], expr.GetType(), isDeferredConst)
+			}
+		} else {
+			isFirstCondition = false
+		}
+		foldedArg1, isDeferred1 := foldConstant(args[i+1])
+		expr.GetArgs()[i+1] = foldedArg1
+		isDeferredConst = isDeferredConst || isDeferred1
+	}
+
+	if l%2 == 1 && isFirstCondition {
+		// If the number of arguments in casewhen is odd, and the previous conditions
+		// is const and false, then the folded else execution body is returned. otherwise
+		// the execution body of the else are folded and replaced.
+		return retProcess(args[l-1], expr.GetType(), isDeferredConst)
+	} else if l%2 == 1 {
+		foldedArg, isDeferred := foldConstant(args[l-1])
```

Comment:
s/ foldedArg/ expr.GetArgs[l-1]

Reply:
‘isDeferred‘  is a new variable here。if s/ foldedArg/ expr.GetArgs[l-1]，I need to declare this variable.

Reply:
‘isDeferred‘ is a new variable here。if s/ foldedArg/ expr.GetArgs[l-1]，I need to declare this variable.

Reply:
We can define isDeferred in line 84.